### DISCUSSION TITLE
Self Hosting - Problems with local deployment using Docker

### DISCUSSION THREAD

**Discussion by ChuckNoelke** [reflex-dev/reflex#2054] [Q&A]
Hi, I am trying to locally run the app in a container as described in the docker-test-folder.

The manual states, that API_URL shall be set to the publicly accessible hostname or IP where the app will be hosted.

What does that mean if I run the container locally? I tried already some options (http://localhost:8000, ip-address of the local container 172.17.0.2:8000, or just used the default). Nothing worked.

Also, I get this warning when starting up the container:

"Warning: The path to the Node binary could not be found. Please ensure that Node
is properly installed and added to your system's PATH environment variable."

Do I need the to adjust the provided Dockerfile in this repo under Docker-Test to get Node installed during the build process?

Thanks for any help,
Marcus

**Comment by masenf** [comment]
That warning can be ignored (https://github.com/reflex-dev/reflex/issues/1778). Reflex installs node itself, but the first time it runs (like in a fresh container), it prints this warning because it hasn't installed it yet.

As for the API_URL, it it sounds like `http://localhost:8000` would be correct for local development, assuming your docker container is forwarding port 8000, like `docker run -p8000:8000 ...`. You should be able to access $API_URL/ping and have it say `"pong"`
